### PR TITLE
Make Flatpak use CMake build framework 3.0

### PR DESCRIPTION
**Pull request by tytan652**
### Description

- Fix unused variable in NVENC native to enable the next point
- Make Flatpak use CMake build framework 3.0
  - Also change CMake build type to RelWithDebInfo which is the preferred build type in Freedesktop SDK environments, dependencies will not be touched in this PR.

### Motivation and Context

Move things forward to deprecate CMake 2.0 and battletest CMake 3.0.

### How Has This Been Tested?

Local build

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by derrod — APPROVED** [review]
Well that was easy.

File: build-aux/com.obsproject.Studio.json [hunk 1]
```diff
@@ -75,15 +75,15 @@
             "buildsystem": "cmake-ninja",
             "builddir": true,
             "config-opts": [
-                "-DCMAKE_BUILD_TYPE=Release",
+                "-DCMAKE_BUILD_TYPE=RelWithDebInfo",
+                "-DOBS_CMAKE_VERSION=3",
```
Comment (RytoEX) [inline build-aux/com.obsproject.Studio.json:79]: @PatTheMav does it matter if this is `3` or `3.0.0`?

**Review by RytoEX — COMMENTED** [review]
Looks fine, though I have one nit/question for @PatTheMav .

File: build-aux/com.obsproject.Studio.json (hunk 1 above)
Comment (PatTheMav) [reply build-aux/com.obsproject.Studio.json:79]: It should work nonetheless as long as CMake treats `3` identically to version `3.0.0`. I don't expect us to use any minor or patch versions anyway, so for the time being (and until legacy is removed) I'd be fine with it.

File: build-aux/com.obsproject.Studio.json [hunk 2]
```diff
@@ -75,15 +75,15 @@
             "buildsystem": "cmake-ninja",
             "builddir": true,
             "config-opts": [
-                "-DCMAKE_BUILD_TYPE=Release",
+                "-DCMAKE_BUILD_TYPE=RelWithDebInfo",
```
Comment (PatTheMav) [inline build-aux/com.obsproject.Studio.json:78]: What's the reason to use a less-optimised build config for Flatpak with this change?

File: build-aux/com.obsproject.Studio.json (hunk 1 above)
Comment (RytoEX) [reply build-aux/com.obsproject.Studio.json:79]: Fine with me then.

**Review by RytoEX — APPROVED** [review]

**Pull request merged**